File: use-cases/easy-ship/code/java/src/main/java/lambda/CreateScheduledPackageHandler.java
package lambda;

import com.amazonaws.services.lambda.runtime.Context;
import com.amazonaws.services.lambda.runtime.LambdaLogger;
import com.amazonaws.services.lambda.runtime.RequestHandler;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.google.gson.Gson;
import io.swagger.client.api.EasyShipApi;
import io.swagger.client.model.easyship.Item;
import io.swagger.client.model.easyship.Items;
import io.swagger.client.model.easyship.ModelPackage;
import io.swagger.client.model.easyship.OrderItemSerialNumbers;
import io.swagger.client.model.easyship.CreateScheduledPackageRequest;
import io.swagger.client.model.easyship.PackageDetails;
import lambda.utils.ApiUtils;
import lambda.utils.EasyShipOrderItem;
import lambda.utils.StateMachineInput;

public class CreateScheduledPackageHandler implements RequestHandler<StateMachineInput, StateMachineInput> {

    public StateMachineInput handleRequest(StateMachineInput input, Context context) {
        LambdaLogger logger = context.getLogger();
        logger.log("CreateScheduledPackage Lambda input: " + new Gson().toJson(input));

        try {
            Items items = new Items();
            for (EasyShipOrderItem orderItem : input.getEasyShipOrder().getOrderItems()) {
                logger.log("EasyShipOrderItem : " + new Gson().toJson(orderItem));
                Item item = new Item();
                item.setOrderItemId(orderItem.getOrderItemId());
                // Set the SerialNumber if it exists, as some region requires this value.
                if (orderItem.getOrderItemSerialNumbers() != null && !orderItem.getOrderItemSerialNumbers().isEmpty()) {
                    item.setOrderItemSerialNumbers((OrderItemSerialNumbers) orderItem.getOrderItemSerialNumbers());
                }
                items.add(item);
            }
            //Get list of handover shipment slot for the order
            CreateScheduledPackageRequest request = new CreateScheduledPackageRequest()
                    .amazonOrderId(input.getAmazonOrderId())
                    .marketplaceId(input.getMarketplaceId())
                    .packageDetails(new PackageDetails()
                            .packageTimeSlot(input.getTimeSlots().get(0)));

            logger.log("EasyShip API -  CreateScheduledPackage request: " + new Gson().toJson(request));

            EasyShipApi easyShipApi = ApiUtils.getEasyShipApi(input);
            ModelPackage response = easyShipApi.createScheduledPackage(request);
            logger.log("EasyShip API -  CreateScheduledPackage response: " + new Gson().toJson(response));
            // Store ScheduledPackageId to validate scheduled correctly using this information on the next step
            input.setScheduledPackageId(response.getScheduledPackageId());
            return input;
        } catch (JsonProcessingException e) {
            throw new InternalError("Message body could not be mapped to EasyShipOrder", e);
        } catch (Exception e) {
            throw new InternalError("Calling EasyShip API failed", e);
        }
    }
}

File: use-cases/easy-ship/code/php/lambda/CreateScheduledPackageHandler.php
<?php

namespace Lambda;

use GuzzleHttp\Exception\RequestException;
use Lambda\Utils\ApiUtils;
use Lambda\Utils\LambdaContext;
use Lambda\Utils\Model\StateMachineInput;
use Lambda\Utils\StateMachineInputConverter;
use SpApi\Model\easyship\v2022_03_23\CreateScheduledPackageRequest;
use SpApi\Model\easyship\v2022_03_23\Item;
use SpApi\Model\easyship\v2022_03_23\PackageDetails;
use SpApi\Model\easyship\v2022_03_23\TimeSlot;

class CreateScheduledPackageHandler
{
    /**
     * @throws \Exception
     */
    public function handleRequest(array $event, LambdaContext $context): StateMachineInput
    {
        $input = StateMachineInputConverter::convertFromArray($event);
        $logger = $context->getLogger();
        $logger->info('CreateScheduledPackage Lambda input: ' . json_encode($event, JSON_PRETTY_PRINT));
        $logger->info('CreateScheduledPackage Lambda MappedInput: ' . json_encode($input));

        try {
            $items = [];
            foreach ($input->getEasyShipOrder()->getOrderItems() as $orderItem) {
                $logger->info('EasyShipOrderItem: ' . json_encode($orderItem));
                $item = new Item();
                $item->setOrderItemId($orderItem->getOrderItemId());
                // Set the SerialNumber if it exists, as some region requires this value
                if (!empty($orderItem->getOrderItemSerialNumbers())) {
                    $item->setOrderItemSerialNumbers($orderItem->getOrderItemSerialNumbers());
                }
                $items[] = $item;
            }

            // Prepare the CreateScheduledPackageRequest payload
            $requestPayload = (new CreateScheduledPackageRequest())
                ->setAmazonOrderId($input->getAmazonOrderId())
                ->setMarketplaceId($input->getMarketplaceId())
                ->setPackageDetails(
                    (new PackageDetails())
                        ->setPackageTimeSlot($input->getTimeSlots()[0])
                        ->setPackageItems($items)
                );
            $logger->info('EasyShip API - CreateScheduledPackage request: ' . json_encode($requestPayload));

            // Create an API instance
            $easyShipApi = ApiUtils::getEasyShipApi($input);
            // Call the EasyShip API
            $response = $easyShipApi->createScheduledPackage($requestPayload);

            $logger->info('EasyShip API - CreateScheduledPackage response: ' . json_encode($response));

            // Store ScheduledPackageId to validate scheduled correctly using this information on the next step
            $input->setScheduledPackageId($response->getScheduledPackageId());

            return $input;
        } catch (RequestException $e) {
            $logger->error('RequestException: ' . $e->getMessage());
            throw new \Exception('Calling EasyShip API failed', 0, $e);
        } catch (\Exception $e) {
            $logger->error('Exception: ' . $e->getMessage());
            throw $e;
        }
    }
}
